can-debug.js

```diff
@@ -2,34 +2,76 @@ var namespace = require('can-namespace');
 var canSymbol = require('can-symbol');
 var canReflect = require('can-reflect');
 
-var getValueDependenciesSymbol = canSymbol.for('can.getValueDependencies');
-
-function debug(obj) {
-  var debugData = {
-    cid: obj._cid,
-    obj: obj,
-    value: canReflect.getValue(obj),
-    valueDependencies: [],
-    keyDependencies: {}
-  };
-
-  if (obj[getValueDependenciesSymbol]) {
-    var deps = canReflect.getValueDependencies(obj) || {};
-
-    if (deps.valueDependencies) {
-      deps.valueDependencies.forEach(function(valueDep) {
-        debugData.valueDependencies.push( debug(valueDep) );
-      });
-    }
-
-    if (deps.keyDependencies) {
-      deps.keyDependencies.forEach(function(events, key) {
-        debugData.keyDependencies[events.join(',')] = debug(key);
-      });
-    }
-  }
-
-  return debugData;
+function debug(obj, key) {
+	// key can be 0, maybe undefined
+	var gotKey = arguments.length === 2;
+
+	var data = {
+		obj: obj,
+		key: key,
+		keyDependencies: {},
+		valueDependencies: [],
+		name: canReflect.getName(obj),
+		value: gotKey ? canReflect.getKeyValue(obj, key) : canReflect.getValue(obj)
+	};
+
+	var deps = gotKey ? getKeyDependencies(obj, key) : getValueDependencies(obj);
+	if (!deps) {
+		return data;
+	}
+
+	// deps.keyDependencies :: Map
+	if (deps.keyDependencies) {
+		canReflect.each(deps.keyDependencies, function(value, obj) {
+			canReflect.each(value, function(key) {
+				data.keyDependencies[key] = debug(obj, key);
+			});
+		});
+	}
+
+	// deps.valueDependencies :: Set
+	if (deps.valueDependencies) {
+		canReflect.each(deps.valueDependencies, function(obj) {
+			data.valueDependencies.push(debug(obj));
+		});
+	}
+
+	return data;
+}
+
+debug.logWhatChangesMe = function(obj, key) {
+	var quoteString = function(x) {
+		return typeof x === 'string' ? JSON.stringify(x) : x;
+	};
+
+	var log = function log(data) {
+		var nameParts = [ data.name, data.key != null ? ('.' + data.key) : ''];
+
+		console.group(nameParts.join(''));
+		console.log('value  ', quoteString(data.value));
+		console.log('object ', data.obj);
+		canReflect.each(data.keyDependencies, log);
+		canReflect.each(data.valueDependencies, log);
+		console.groupEnd();
+	};
+
+	log(debug(obj, key));
+};
+
+function getKeyDependencies(obj, key) {
+	var getKeyDependenciesSymbol = canSymbol.for('can.getKeyDependencies');
```

Comment:
cache the key outside the function so you don't have to call `for` so many times.